### PR TITLE
Switch build system to hatchling and add noxfile

### PR DESCRIPTION
**Pull request by felixfontein**

### CODE REVIEW

**Comment by gotmax23** [comment]
I had some work on this locally that I applied on top of your PR: https://github.com/gotmax23/antsibull-changelog/commits/build-system. Feel free to pull it in and make any changes you see fit or I can push it to your fork myself.

**Comment by gotmax23** [comment]
Also, I'm confused as to why there's a .mypy and mypy.ini. We might want to consider switching to the pyproject.toml based mypy configuration in a separate PR.

File: pyproject.toml [hunk 1]
```diff
@@ -3,56 +3,83 @@
...
-pyre-check = "^0.9.17"
-pylint = "*"
-mypy = "*"
-mock = "*"
-types-docutils = "*"
-types-PyYAML = "*"
-types-toml = "*"
+[project.optional-dependencies]
+test = [
+    "mock",
```
Comment (gotmax23) [inline pyproject.toml:57]: Note to self: we should get rid of the `mock` dependency and use `unittest.mock` instead. The `mock` package is a backport of stdlib code that's not necessary for newer pythons.

**Comment by felixfontein** [comment]
> Also, I'm confused as to why there's a .mypy and mypy.ini. We might want to consider switching to the pyproject.toml based mypy configuration in a separate PR.

There should have only been `.mypy.ini`. I agree on consolidating to use pyproject.toml for all the other tools (which support it), having all these random small config files around is annoying :)

**Review by gotmax23 — APPROVED** [review]
I have two suggestions, but this looks good to me. Thanks for working on it!

**Comment by felixfontein** [comment]
@gotmax23 thanks a lot for preparing this and reviewing :)

**Pull request merged**